mypy.ini

```diff
@@ -22,7 +22,6 @@ follow_imports = silent
 files =
     torch,
     caffe2,
-    test/type_hint_tests,
```

Comment:
I think I'm missing some context; why is this line being removed?

Reply:
That whole directory is being removed. All the files in it test very little, and we now have a much better way of testing with the "reveal" style tests under `test/typing/`.

Reply:
Ah ok thanks! Followup question: how does this relate to #50513? Like I see that now we have two different test files (`test/test_type_hints.py` and `test/test_typing.py`) that both run `mypy`

Reply:
Those two files do different things: the first one just runs mypy over the whole code base, the second one has specific unit tests for individual annotations. I'm not sure I see a compelling reason to want to run them in a single command. It's possible of course, but `mypy` is just a tool and type annotations are just one code aspect - we also don't say we want to run (e.g.) all autograd tests in a single command. There's just tests scattered in places.

The one thing that would be good is better mypy cache reuse. Right now `test_type_hints.py` can generate caches in the repo root and in `test/`, and `test_typing.py` in `test/typing/`. It'd be nice if all caches for the same `.ini` file ended up only in the repo root.

Reply:
Makes sense :) Regarding cache reuse, though, would that necessarily be better if they all shared the same cache dir? It seems like that might provide opportunity for additional unnecessary cache invalidation, since the commands being invoked are somewhat different from each other.

Reply:
True, I'm not quite sure if it would be better. As long as the regression tests only take 10 seconds or so (which is the case now), it's not even worth optimizing I think.